x/market/genesis.go

```diff
@@ -22,14 +22,34 @@ func InitGenesis(ctx sdk.Context, k keeper.Keeper, genState types.GenesisState)
 
 // ExportGenesis returns the capability module's exported genesis.
 func ExportGenesis(ctx sdk.Context, k keeper.Keeper) *types.GenesisState {
-	//genesis := types.DefaultGenesis()
+	deals, err := k.ListDeals(ctx)
+	if err != nil {
+		panic(err)
+	}
 
 	dealsMap := make(map[uint64]*types.Deal)
+	for _, deal := range deals {
+		deal, err := k.GetDeal(ctx, deal.DealId)
+		if err != nil {
+			panic(err)
+		}
+		dealsMap[deal.DealId] = &deal
+	}
+
+	dataCertificates, err := k.ListDataCertificates(ctx)
+	if err != nil {
+		panic(err)
+	}
 
 	dataCertificateMap := make(map[string]*types.DataValidationCertificate)
-	//TODO: Implement GetDealsList and GetDataCertificateByDealId Mapping to dealsMap and dataCertificateMap.
-	//for _, deal := k.GetDeals(ctx) {
-	//}
+	for _, dataCertificate := range dataCertificates {
+		certificate, err := k.GetDataCertificate(ctx, dataCertificate)
+		if err != nil {
+			panic(err)
+		}
+		dataKey := types.CombineKeys(sdk.Uint64ToBigEndian(certificate.UnsignedCert.DealId), certificate.UnsignedCert.DataHash)
+		dataCertificateMap[string(dataKey)] = &certificate
```

Comment:
```suggestion
		dataKey := types.CombineKeys(sdk.Uint64ToBigEndian(dataCertificate.UnsignedCert.DealId), dataCertificate.UnsignedCert.DataHash)
		dataCertificateMap[string(dataKey)] = &dataCertificate
```

I think you can write it like this.